onnxruntime/core/framework/tensorprotoutils.cc

```diff
@@ -165,23 +165,57 @@ static Status GetExternalDataInfo(const ONNX_NAMESPACE::TensorProto& tensor_prot
 // This function does not unpack string_data of an initializer tensor
 Status ReadExternalDataForTensor(const ONNX_NAMESPACE::TensorProto& tensor_proto,
                                  const ORTCHAR_T* tensor_proto_dir,
-                                 std::vector<uint8_t>& unpacked_tensor) {
+                                 std::unique_ptr<uint8_t[]>& unpacked_tensor, size_t& tensor_byte_size) {
   std::basic_string<ORTCHAR_T> external_file_path;
   onnxruntime::FileOffsetType file_offset;
-  SafeInt<size_t> tensor_byte_size;
+  SafeInt<size_t> safe_tensor_byte_size;
   ORT_RETURN_IF_ERROR(GetExternalDataInfo(
       tensor_proto,
       tensor_proto_dir,
       external_file_path,
       file_offset,
-      tensor_byte_size));
+      safe_tensor_byte_size));
+  tensor_byte_size = safe_tensor_byte_size;
 
-  unpacked_tensor.resize(tensor_byte_size);
+  unpacked_tensor = std::unique_ptr<uint8_t[]>(new uint8_t[tensor_byte_size]); // std::make_unique_for_overwrite<uint8_t[]>(tensor_byte_size);
```

Comment:
Can we remove the comment here? Or did you want to replace `std::unique_ptr<uint8_t[]>(new uint8_t[tensor_byte_size])` with it?

Reply:
Once C++20 is available I'd like to use `std::make_unique_for_overwrite` since it allocates memory without initializing it. `std::make_unqiue(new)` is not recommended due to exception safety and thus bad programming style. `std::make_unique<>(size)` will initialize data and thus introduce the memset.

Reply:
A TODO here wouldn't hurt, to make sure that we don't misinterpret this comment as dead code and remember to change it once C++20 becomes available.